src/viam/sdk/services/mlmodel/client.cpp

```diff
@@ -69,7 +69,7 @@ std::shared_ptr<MLModelService::named_tensor_views> MLModelServiceClient::infer(
         aav->views.emplace(kv.first, std::move(tensor));
     }
     auto* const tsav_views = &aav->views;
-    return {std::move(aav), tsav_views};
+    return {aav, tsav_views};
```

Comment:
Surprising, but actually correct. The shared_ptr aliasing constructor takes its first arg by `const&`! I did not know that. Apparently they added the one taking an rvalue reference for C++20. So the code would be correct there. I'm inclined to NOLINT this one with a comment. The move is harmless up to C++17 since no move will happen, but desirable for C++20 where it saves a refcount jiggle.

Reply:
Speaking of: what C++ version is clang tidy using to generate its warnings? I suppose it's not C++20 given this comment, but was curious if it uses C++14 because of the cmakelists value?